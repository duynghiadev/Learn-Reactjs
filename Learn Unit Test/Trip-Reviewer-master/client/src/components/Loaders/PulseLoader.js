import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import assign from 'domkit/appendVendorPrefix';
import insertKeyframesRule from 'domkit/insertKeyframesRule';

const keyframes = {
  '0%': {
    transform: 'scale(1)',
    opacity: 1,
  },
  '45%': {
    transform: 'scale(0.1)',
    opacity: 0.7,
  },
  '80%': {
    transform: 'scale(1)',
    opacity: 1,
  },
};

const animationName = insertKeyframesRule(keyframes);

class PulseLoader extends PureComponent {
  static getAnimationStyle(param) {
    const animation = [
      animationName,
      '0.75s',
      `${param * 0.12}s`,
      'infinite',
      'cubic-bezier(.2,.68,.18,1.08)',
    ].join(' ');
    const animationFillMode = 'both';

    return {
      animation,
      animationFillMode,
    };
  }

  getBallStyle() {
    return {
      backgroundColor: this.props.color || 'grey',
      width: this.props.size || '15px',
      height: this.props.size || '15px',
      margin: this.props.margin || '2px',
      borderRadius: '100%',
      verticalAlign: this.props.verticalAlign,
    };
  }

  getStyle(param) {
    return assign(this.getBallStyle(param), PulseLoader.getAnimationStyle(param), {
      display: 'inline-block',
    });
  }

  render() {
    return (
      <div id={this.props.id} className={this.props.className}>
        <div style={this.getStyle(1)} />
        <div style={this.getStyle(2)} />
        <div style={this.getStyle(3)} />
      </div>
    );
  }
}

PulseLoader.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
  margin: PropTypes.string,
  verticalAlign: PropTypes.string,
};

PulseLoader.defaultProps = {
  id: '',
  className: '',
  color: '',
  size: '',
  margin: '',
  verticalAlign: undefined,
};

export default PulseLoader;
