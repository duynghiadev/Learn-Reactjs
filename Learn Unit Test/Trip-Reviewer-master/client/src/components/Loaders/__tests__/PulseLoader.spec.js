import React from 'react';
import { create } from 'react-test-renderer';

import PulseLoader from '../PulseLoader';

jest.mock('domkit/insertKeyframesRule', () => jest.fn(() => 'animation'));

const mockComponent = props => <PulseLoader {...props} />;

describe('<PulseLoader />', () => {
  it('should render component with default props', () => {
    const props = {};
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with custom props', () => {
    const props = {
      color: 'color',
      size: 'size',
      margin: 'margin',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
