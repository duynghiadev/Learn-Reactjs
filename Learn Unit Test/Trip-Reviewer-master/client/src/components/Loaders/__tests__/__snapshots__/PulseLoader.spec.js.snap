// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PulseLoader /> should render component with custom props 1`] = `
<div
  className=""
  id=""
>
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.12s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "color",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "size",
        "margin": "margin",
        "verticalAlign": undefined,
        "width": "size",
      }
    }
  />
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.24s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "color",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "size",
        "margin": "margin",
        "verticalAlign": undefined,
        "width": "size",
      }
    }
  />
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.36s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "color",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "size",
        "margin": "margin",
        "verticalAlign": undefined,
        "width": "size",
      }
    }
  />
</div>
`;

exports[`<PulseLoader /> should render component with default props 1`] = `
<div
  className=""
  id=""
>
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.12s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "grey",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "15px",
        "margin": "2px",
        "verticalAlign": undefined,
        "width": "15px",
      }
    }
  />
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.24s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "grey",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "15px",
        "margin": "2px",
        "verticalAlign": undefined,
        "width": "15px",
      }
    }
  />
  <div
    style={
      Object {
        "animation": "animation 0.75s 0.36s infinite cubic-bezier(.2,.68,.18,1.08)",
        "animationFillMode": "both",
        "backgroundColor": "grey",
        "borderRadius": "100%",
        "display": "inline-block",
        "height": "15px",
        "margin": "2px",
        "verticalAlign": undefined,
        "width": "15px",
      }
    }
  />
</div>
`;
